projects/public-search/src/app/document-detail/request/pickup-location/pickup-location.component.ts

```diff
@@ -98,20 +121,30 @@ export class PickupLocationComponent implements OnInit {
   /** Submit form */
   submit() {
     this.requestInProgress = true;
-    this._itemApiService.request({
-      item_pid: this.item.metadata.pid,
-      pickup_location_pid: this.model.pickup,
-    })
-    .pipe(tap(() => {
+    if (this.recordType === 'holding') {
+      this.apiRequest = this._holdingsApiService.request({
+        holding_pid: this.record.metadata.pid,
+        pickup_location_pid: this.model.pickup,
+        description: this.model.description,
+      });
+    } else if (this.recordType === 'item') {
+      this.apiRequest = this._itemApiService.request({
+        item_pid: this.record.metadata.pid,
+        pickup_location_pid: this.model.pickup,
+      });
+    }
+
+    this.apiRequest.pipe(tap(() => {
       this.showForm = false;
       this.requestInProgress = false;
       this.requested = true;
     }))
     .subscribe(
       () => {
+        // let message = {'message_key': 'A request has been placed on this {{ type }}'};
         this.requestMessage = {
           success: true,
-          message: this._translateService.instant('A request has been placed on this item.')
+          message: this._translateService.instant(`A request has been placed on this ${ this.recordType }`)
```

Comment:
`this._translateService.instant('A request has been placed on this {type}', { type: this.recordType })`